// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/interfaces/INativeCurrencyPaymentFallback__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.INativeCurrencyPaymentFallback__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "sender",
                type: "address",
            },
            {
                indexed: false,
                internalType: "uint256",
                name: "value",
                type: "uint256",
            },
        ],
        name: "SafeReceived",
        type: "event",
    },
    {
        stateMutability: "payable",
        type: "receive",
    },
];
class INativeCurrencyPaymentFallback__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.INativeCurrencyPaymentFallback__factory = INativeCurrencyPaymentFallback__factory;
INativeCurrencyPaymentFallback__factory.abi = _abi;
