// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/FallbackHandler.sol/FallbackHandler__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FallbackHandler__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "selector",
                type: "bytes4",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "oldMethod",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "newMethod",
                type: "bytes32",
            },
        ],
        name: "ChangedSafeMethod",
        type: "event",
    },
    {
        stateMutability: "nonpayable",
        type: "fallback",
    },
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "",
                type: "bytes4",
            },
        ],
        name: "safeMethods",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "selector",
                type: "bytes4",
            },
            {
                internalType: "bytes32",
                name: "newMethod",
                type: "bytes32",
            },
        ],
        name: "setSafeMethod",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class FallbackHandler__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.FallbackHandler__factory = FallbackHandler__factory;
FallbackHandler__factory.abi = _abi;
