// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/FallbackHandler.sol/IFallbackHandler__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IFallbackHandler__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "selector",
                type: "bytes4",
            },
            {
                internalType: "bytes32",
                name: "newMethod",
                type: "bytes32",
            },
        ],
        name: "setSafeMethod",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class IFallbackHandler__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.IFallbackHandler__factory = IFallbackHandler__factory;
IFallbackHandler__factory.abi = _abi;
