// @safe-global/safe-smart-account/dist/test/utils/setup.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAbi = exports.getSignMessageLib = exports.deployContractFromSource = exports.compile = exports.getDelegateCaller = exports.getSafeProxyRuntimeCode = exports.getExtensibleFallbackHandler = exports.getCompatFallbackHandler = exports.getTokenCallbackHandler = exports.getSafe = exports.getSafeTemplateWithSingleton = exports.getSafeTemplate = exports.getMock = exports.safeMigrationContract = exports.migrationContractFactory = exports.getCreateCall = exports.getMultiSendCallOnly = exports.getMultiSend = exports.getSimulateTxAccessor = exports.getFactory = exports.getSafeSingletonAt = exports.getSafeL2Singleton = exports.getSafeL1Singleton = exports.getSafeSingleton = exports.defaultTokenCallbackHandlerDeployment = void 0;
const hardhat_1 = __importStar(require("hardhat"));
const ethers_1 = require("ethers");
const constants_1 = require("@ethersproject/constants");
const solc_1 = __importDefault(require("solc"));
const execution_1 = require("../../src/utils/execution");
const config_1 = require("./config");
const numbers_1 = require("./numbers");
const defaultTokenCallbackHandlerDeployment = async () => {
    return hardhat_1.deployments.get("TokenCallbackHandler");
};
exports.defaultTokenCallbackHandlerDeployment = defaultTokenCallbackHandlerDeployment;
const getSafeSingleton = async () => {
    const safeContractName = (0, config_1.safeContractUnderTest)();
    const { address } = await hardhat_1.deployments.get(safeContractName);
    const safe = await hardhat_1.default.ethers.getContractAt(safeContractName, address);
    return safe;
};
exports.getSafeSingleton = getSafeSingleton;
const getSafeL1Singleton = async () => {
    const safeSingletonDeployment = await hardhat_1.deployments.get("Safe");
    const Safe = await hardhat_1.default.ethers.getContractAt("Safe", safeSingletonDeployment.address);
    return Safe;
};
exports.getSafeL1Singleton = getSafeL1Singleton;
const getSafeL2Singleton = async () => {
    const safeSingletonDeployment = await hardhat_1.deployments.get("SafeL2");
    const Safe = await hardhat_1.default.ethers.getContractAt("SafeL2", safeSingletonDeployment.address);
    return Safe;
};
exports.getSafeL2Singleton = getSafeL2Singleton;
const getSafeSingletonAt = async (address) => {
    const safe = await hardhat_1.default.ethers.getContractAt((0, config_1.safeContractUnderTest)(), address);
    return safe;
};
exports.getSafeSingletonAt = getSafeSingletonAt;
const getFactory = async (address) => {
    if (!address) {
        const factoryDeployment = await hardhat_1.deployments.get("SafeProxyFactory");
        address = factoryDeployment.address;
    }
    const Factory = await hardhat_1.default.ethers.getContractAt("SafeProxyFactory", address);
    return Factory;
};
exports.getFactory = getFactory;
const getSimulateTxAccessor = async () => {
    const SimulateTxAccessor = await hardhat_1.default.ethers.getContractAt("SimulateTxAccessor", (await hardhat_1.deployments.get("SimulateTxAccessor")).address);
    return SimulateTxAccessor;
};
exports.getSimulateTxAccessor = getSimulateTxAccessor;
const getMultiSend = async () => {
    const MultiSend = await hardhat_1.default.ethers.getContractAt("MultiSend", (await hardhat_1.deployments.get("MultiSend")).address);
    return MultiSend;
};
exports.getMultiSend = getMultiSend;
const getMultiSendCallOnly = async () => {
    const MultiSend = await hardhat_1.default.ethers.getContractAt("MultiSendCallOnly", (await hardhat_1.deployments.get("MultiSendCallOnly")).address);
    return MultiSend;
};
exports.getMultiSendCallOnly = getMultiSendCallOnly;
const getCreateCall = async () => {
    const CreateCall = await hardhat_1.default.ethers.getContractAt("CreateCall", (await hardhat_1.deployments.get("CreateCall")).address);
    return CreateCall;
};
exports.getCreateCall = getCreateCall;
const migrationContractFactory = async () => {
    return await hardhat_1.default.ethers.getContractFactory("Migration");
};
exports.migrationContractFactory = migrationContractFactory;
const safeMigrationContract = async () => {
    const safeMigration = await hardhat_1.default.ethers.getContractAt("SafeMigration", (await hardhat_1.deployments.get("SafeMigration")).address);
    return safeMigration;
};
exports.safeMigrationContract = safeMigrationContract;
const getMock = async () => {
    const contractFactory = await hardhat_1.default.ethers.getContractFactory("MockContract");
    const contract = await contractFactory.deploy();
    return contract;
};
exports.getMock = getMock;
const getSafeTemplate = async (saltNumber = (0, numbers_1.getRandomIntAsString)()) => {
    const singleton = await (0, exports.getSafeSingleton)();
    return (0, exports.getSafeTemplateWithSingleton)(singleton, saltNumber);
};
exports.getSafeTemplate = getSafeTemplate;
const getSafeTemplateWithSingleton = async (singleton, saltNumber = (0, numbers_1.getRandomIntAsString)()) => {
    const singletonAddress = await singleton.getAddress();
    const factory = await (0, exports.getFactory)();
    const template = await factory.createProxyWithNonce.staticCall(singletonAddress, "0x", saltNumber);
    await factory.createProxyWithNonce(singletonAddress, "0x", saltNumber).then((tx) => tx.wait());
    return singleton.attach(template);
};
exports.getSafeTemplateWithSingleton = getSafeTemplateWithSingleton;
const getSafe = async (safe) => {
    const { singleton = await (0, exports.getSafeSingleton)(), owners, threshold = owners.length, to = constants_1.AddressZero, data = "0x", fallbackHandler = constants_1.AddressZero, logGasUsage = false, saltNumber = (0, numbers_1.getRandomIntAsString)(), } = safe;
    const template = await (0, exports.getSafeTemplateWithSingleton)(singleton, saltNumber);
    await (0, execution_1.logGas)(`Setup Safe with ${owners.length} owner(s)${fallbackHandler && fallbackHandler !== constants_1.AddressZero ? " and fallback handler" : ""}`, template.setup(owners, threshold, to, data, fallbackHandler, constants_1.AddressZero, 0, constants_1.AddressZero), !logGasUsage);
    return template;
};
exports.getSafe = getSafe;
const getTokenCallbackHandler = async (address) => {
    if (!address) {
        const tokenCallbackHandlerDeployment = await (0, exports.defaultTokenCallbackHandlerDeployment)();
        address = tokenCallbackHandlerDeployment.address;
    }
    const tokenCallbackHandler = await hardhat_1.default.ethers.getContractAt("TokenCallbackHandler", address);
    return tokenCallbackHandler;
};
exports.getTokenCallbackHandler = getTokenCallbackHandler;
const getCompatFallbackHandler = async (address) => {
    if (!address) {
        const fallbackHandlerDeployment = await hardhat_1.deployments.get("CompatibilityFallbackHandler");
        address = fallbackHandlerDeployment.address;
    }
    const fallbackHandler = await hardhat_1.default.ethers.getContractAt("CompatibilityFallbackHandler", address);
    return fallbackHandler;
};
exports.getCompatFallbackHandler = getCompatFallbackHandler;
const getExtensibleFallbackHandler = async (address) => {
    if (!address) {
        const extensibleFallbackHandlerAddress = await hardhat_1.deployments.get("ExtensibleFallbackHandler");
        address = extensibleFallbackHandlerAddress.address;
    }
    const extensibleFallbackHandler = await hardhat_1.default.ethers.getContractAt("ExtensibleFallbackHandler", address);
    return extensibleFallbackHandler;
};
exports.getExtensibleFallbackHandler = getExtensibleFallbackHandler;
const getSafeProxyRuntimeCode = async () => {
    const proxyArtifact = await hardhat_1.default.artifacts.readArtifact("SafeProxy");
    return proxyArtifact.deployedBytecode;
};
exports.getSafeProxyRuntimeCode = getSafeProxyRuntimeCode;
const getDelegateCaller = async () => {
    const DelegateCaller = await hardhat_1.default.ethers.getContractFactory("DelegateCaller");
    return await DelegateCaller.deploy();
};
exports.getDelegateCaller = getDelegateCaller;
const compile = async (source) => {
    const input = JSON.stringify({
        language: "Solidity",
        settings: {
            outputSelection: {
                "*": {
                    "*": ["abi", "evm.bytecode"],
                },
            },
        },
        sources: {
            "tmp.sol": {
                content: source,
            },
        },
    });
    const solcData = await solc_1.default.compile(input);
    const output = JSON.parse(solcData);
    if (!output["contracts"]) {
        console.log(output);
        throw Error("Could not compile contract");
    }
    const fileOutput = output["contracts"]["tmp.sol"];
    // Find the first contract with bytecode in the output, this allows the
    // compiled code to include interfaces.
    for (const contract in fileOutput) {
        const contractOutput = fileOutput[contract];
        if (!contractOutput["evm"]["bytecode"] || !contractOutput["evm"]["bytecode"]["object"]) {
            continue;
        }
        const abi = contractOutput["abi"];
        const data = "0x" + contractOutput["evm"]["bytecode"]["object"];
        return {
            data,
            interface: abi,
        };
    }
    console.log(output);
    throw Error("No contract with bytecode");
};
exports.compile = compile;
const deployContractFromSource = async (deployer, source) => {
    const output = await (0, exports.compile)(source);
    const transaction = await deployer.sendTransaction({ data: output.data, gasLimit: 6000000 });
    const receipt = await transaction.wait();
    if (!receipt?.contractAddress) {
        throw Error("Could not deploy contract");
    }
    return new ethers_1.Contract(receipt.contractAddress, output.interface, deployer);
};
exports.deployContractFromSource = deployContractFromSource;
const getSignMessageLib = async () => {
    const SignMessageLibDeployment = await hardhat_1.deployments.get("SignMessageLib");
    const SignMessageLib = await hardhat_1.default.ethers.getContractAt("SignMessageLib", SignMessageLibDeployment.address);
    return SignMessageLib;
};
exports.getSignMessageLib = getSignMessageLib;
const getAbi = async (name) => {
    const artifact = await hardhat_1.default.artifacts.readArtifact(name);
    if (!artifact) {
        throw Error(`Could not read artifact for ${name}`);
    }
    return artifact.abi;
};
exports.getAbi = getAbi;
