// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/ERC165Handler.sol/ERC165Handler__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ERC165Handler__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "interfaceId",
                type: "bytes4",
            },
        ],
        name: "AddedInterface",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "selector",
                type: "bytes4",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "oldMethod",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "newMethod",
                type: "bytes32",
            },
        ],
        name: "ChangedSafeMethod",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "interfaceId",
                type: "bytes4",
            },
        ],
        name: "RemovedInterface",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "_interfaceId",
                type: "bytes4",
            },
            {
                internalType: "bytes32[]",
                name: "handlerWithSelectors",
                type: "bytes32[]",
            },
        ],
        name: "addSupportedInterfaceBatch",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "_interfaceId",
                type: "bytes4",
            },
            {
                internalType: "bytes4[]",
                name: "selectors",
                type: "bytes4[]",
            },
        ],
        name: "removeSupportedInterfaceBatch",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "",
                type: "bytes4",
            },
        ],
        name: "safeInterfaces",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "",
                type: "bytes4",
            },
        ],
        name: "safeMethods",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "interfaceId",
                type: "bytes4",
            },
            {
                internalType: "bool",
                name: "supported",
                type: "bool",
            },
        ],
        name: "setSupportedInterface",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes4",
                name: "interfaceId",
                type: "bytes4",
            },
        ],
        name: "supportsInterface",
        outputs: [
            {
                internalType: "bool",
                name: "",
                type: "bool",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class ERC165Handler__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.ERC165Handler__factory = ERC165Handler__factory;
ERC165Handler__factory.abi = _abi;
