// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/SignatureVerifierMuxer.sol/ISafeSignatureVerifier__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISafeSignatureVerifier__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                internalType: "address",
                name: "sender",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "_hash",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "domainSeparator",
                type: "bytes32",
            },
            {
                internalType: "bytes32",
                name: "typeHash",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "encodeData",
                type: "bytes",
            },
            {
                internalType: "bytes",
                name: "payload",
                type: "bytes",
            },
        ],
        name: "isValidSafeSignature",
        outputs: [
            {
                internalType: "bytes4",
                name: "magic",
                type: "bytes4",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
class ISafeSignatureVerifier__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.ISafeSignatureVerifier__factory = ISafeSignatureVerifier__factory;
ISafeSignatureVerifier__factory.abi = _abi;
