// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/SignatureVerifierMuxer.sol/ISignatureVerifierMuxer__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ISignatureVerifierMuxer__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "domainSeparator",
                type: "bytes32",
            },
        ],
        name: "domainVerifiers",
        outputs: [
            {
                internalType: "contract ISafeSignatureVerifier",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "domainSeparator",
                type: "bytes32",
            },
            {
                internalType: "contract ISafeSignatureVerifier",
                name: "verifier",
                type: "address",
            },
        ],
        name: "setDomainVerifier",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class ISignatureVerifierMuxer__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.ISignatureVerifierMuxer__factory = ISignatureVerifierMuxer__factory;
ISignatureVerifierMuxer__factory.abi = _abi;
