// @safe-global/safe-smart-account/dist/typechain-types/factories/contracts/handler/extensible/SignatureVerifierMuxer.sol/SignatureVerifierMuxer__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignatureVerifierMuxer__factory = void 0;
const ethers_1 = require("ethers");
const _abi = [
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "domainSeparator",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "contract ISafeSignatureVerifier",
                name: "oldVerifier",
                type: "address",
            },
            {
                indexed: false,
                internalType: "contract ISafeSignatureVerifier",
                name: "newVerifier",
                type: "address",
            },
        ],
        name: "ChangedDomainVerifier",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "contract ISafe",
                name: "safe",
                type: "address",
            },
            {
                indexed: false,
                internalType: "bytes4",
                name: "selector",
                type: "bytes4",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "oldMethod",
                type: "bytes32",
            },
            {
                indexed: false,
                internalType: "bytes32",
                name: "newMethod",
                type: "bytes32",
            },
        ],
        name: "ChangedSafeMethod",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "",
                type: "address",
            },
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        name: "domainVerifiers",
        outputs: [
            {
                internalType: "contract ISafeSignatureVerifier",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "_hash",
                type: "bytes32",
            },
            {
                internalType: "bytes",
                name: "signature",
                type: "bytes",
            },
        ],
        name: "isValidSignature",
        outputs: [
            {
                internalType: "bytes4",
                name: "magic",
                type: "bytes4",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "contract ISafe",
                name: "",
                type: "address",
            },
            {
                internalType: "bytes4",
                name: "",
                type: "bytes4",
            },
        ],
        name: "safeMethods",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "domainSeparator",
                type: "bytes32",
            },
            {
                internalType: "contract ISafeSignatureVerifier",
                name: "newVerifier",
                type: "address",
            },
        ],
        name: "setDomainVerifier",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
class SignatureVerifierMuxer__factory {
    static createInterface() {
        return new ethers_1.Interface(_abi);
    }
    static connect(address, runner) {
        return new ethers_1.Contract(address, _abi, runner);
    }
}
exports.SignatureVerifierMuxer__factory = SignatureVerifierMuxer__factory;
SignatureVerifierMuxer__factory.abi = _abi;
